queries/lua.scm

```diff
@@ -0,0 +1,260 @@
+;; Statements
+[
+    (variable_declaration)
+    (break_statement)
+    (do_statement)
+    (elseif_statement)
+    (else_statement)
+    (empty_statement)
+    (for_statement)
+    (goto_statement)
+    (if_statement)
+    (label_statement)
+    (repeat_statement)
+    (return_statement)
+    (while_statement)
+] @statement
+
+;; Only treat function declarions and calls as statements if they
+;; aren't part of assignments, etc
+(
+    [
+        (function_declaration)
+        (function_call)
+    ] @statement
+    (#not-parent-type? @statement expression_list)
+)
+[
+    (block)
+    (chunk)
+] @statement.iteration @namedFunction.iteration @functionCall.iteration @ifStatement.iteration
```

Comment:
you'll find that this isn't working for `ifStatement`. Annoyingly, tree-sitter labels silently get ignored after the third label on a single node 😳

Reply:
Huh. So is the solution to have multiple instances of `[ ... ] @...` with up to 3 labels each or there is actually just no way to do it at all as it's a hard underlying limit?

Is that the case if you use `@_.iteration` where `@_` matches over 3 scopes as well?

Will have to do some experimenting...

Reply:
> Huh. So is the solution to have multiple instances of `[ ... ] @...` with up to 3 labels each or there is actually just no way to do it at all as it's a hard underlying limit?

Yeah you can just duplicate the pattern with the remaining captures

> Is that the case if you use `@_.iteration` where `@_` matches over 3 scopes as well?

No that's fine